### PR TITLE
Class to generate error messages for the user.

### PR DESCRIPTION
**Pull request by nilssonemma**

### CODE REVIEW

File: ErrorMessage/index.ts [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+									message = t("The cardholder has blocked this merchant from using the card.")
+									break
+								default:
+									message = this.unknown
+									break
+							}
+							break
+						case "pares":
+						case "card.pares":
+						case "Card.Token":
```
Comment (robinnedergard) [inline ErrorMessage/index.ts:119]: add cases for:
`case "verification":`
`case "card.verification":`

File: ErrorMessage/index.ts [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+						case "card.pan":
+							switch (content.description) {
+								case "Invalid card number.":
+									message = t("The card number is invalid. Please enter a valid number.")
+									break
+								case "Not enrolled.":
+								case "Unsupported card scheme.":
+									message = t("Your card is not supported. Please try another card.")
+									break
+								case "Card expired":
```
Comment (robinnedergard) [inline ErrorMessage/index.ts:93]: Line `case "Card expired":` can be removed now (but keep the correct one containing a dot).

File: ErrorMessage/index.ts [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+								content.description == "Insufficient funds."
+									? t("There is insufficient funds on the card.") + " " + this.getTypeSpecificMessage()
+									: t("The amount limit is exceeded.") + " " + this.getTypeSpecificMessage()
+							break
+						case "card.pan":
+							switch (content.description) {
+								case "Invalid card number.":
+									message = t("The card number is invalid. Please enter a valid number.")
+									break
+								case "Not enrolled.":
```
Comment (robinnedergard) [inline ErrorMessage/index.ts:89]: move `case "Not enrolled.":` to it's own case with text: `t("Your card doesn't support required 3-D Secure verification. Please try another card.")` and remember to add translation.

File: ErrorMessage/index.ts [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+							(details.body.status.message as string).includes("pan")
+								? t("The card number is invalid. Please enter a valid number.")
+								: t("The card details are incorrect. Please enter valid card details and try again.")
+					else
+						message = this.unknown
+					break
+				case "unknown error":
+					message =
+						event.detail.error && event.detail.error.code == "50092"
+							? t("The card details are incorrect. Please enter valid card details and try again.")
```
Comment (robinnedergard) [inline ErrorMessage/index.ts:167]: Keep for now, but shouldn't be needed in the future.

**Review by robinnedergard — CHANGES_REQUESTED** [review]
Some added fields for compatibility with future changes also requested.

File: ErrorMessage/translation/sv.ts [hunk 5]
```diff
@@ -31,4 +31,6 @@ export const sv = {
 		"Ett okänt fel uppstod. Kontrollera kortuppgifterna och försök igen.",
 	"The account could not be created.": "Kunde inte skapa konto.",
 	"The payment could not be completed.": "Betalningen kunde inte genomföras.",
+	"Your card doesn't support required 3-D Secure verification. Please try another card.":
```
Comment (robinnedergard) [inline ErrorMessage/translation/sv.ts:34]: Noticed different spelling of "3-D Secure" in two other error messages, change all of them to either "3D secure" or "3-D Secure" for consistency.

File: ErrorMessage/index.ts [hunk 6]
```diff
@@ -0,0 +1,187 @@
...
+						case "card.pan":
+							switch (content.description) {
+								case "Invalid card number.":
+									message = t("The card number is invalid. Please enter a valid number.")
+									break
+								case "Unsupported card scheme.":
+									message = t("Your card is not supported. Please try another card.")
+									break
+								case "Not enrolled.":
+									message = t("Your card doesn't support required 3-D Secure verification. Please try another card.")
```
Comment (robinnedergard) [inline ErrorMessage/index.ts:93]: Noticed different spelling of "3-D Secure" in two other error messages, change all of them to either "3D secure" or "3-D Secure" for consistency.

**Review by robinnedergard — CHANGES_REQUESTED** [review]

**Review by robinnedergard — APPROVED** [review]

**Review by fredrikbryntesson — APPROVED** [review]

**Pull request merged**